spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/Recurrent.scala

```diff
@@ -37,23 +36,23 @@ import scala.reflect.ClassTag
 class Recurrent[T : ClassTag](var batchNormParams: BatchNormParams[T] = null)
   (implicit ev: TensorNumeric[T]) extends Container[Tensor[T], Tensor[T], T] {
 
-  private var hidden: Activity = null
-  private var gradHidden: Activity = null
-  private var hiddenShape: Array[Int] = null
-  private val currentInput = T()
-  private val currentGradOutput = T()
-  private val gradInputCell = Tensor[T]()
-  private var outputCell = Tensor[T]()
-  private val _input = T()
-  private val batchDim = Recurrent.batchDim
-  private val timeDim = Recurrent.timeDim
-  private val inputDim = 1
-  private val hidDim = 2
-  private var (batchSize, times) = (0, 0)
-  private var topology: Cell[T] = null
-  private val outputBuffer = Tensor[T]()
+  protected var hidden: Activity = null
+  protected var gradHidden: Activity = null
+  protected var hiddenShape: Array[Int] = null
+  protected val currentInput = T()
+  protected val currentGradOutput = T()
+  protected val gradInputCell = Tensor[T]()
+  protected var outputCell = Tensor[T]()
+  protected var _input = T()
+  protected val batchDim = Recurrent.batchDim
+  protected val timeDim = Recurrent.timeDim
+  protected val inputDim = 1
+  protected val hidDim = 2
+  protected var (batchSize, times) = (0, 0)
+  protected var topology: Cell[T] = null
+  protected val outputBuffer = Tensor[T]()
```

Comment:
the name is confusing - it should be the buffer for the input of a single time step

Reply:
Rename to stepInput2CellBuf